### PR TITLE
Add Camera shake functionality

### PR DESCRIPTION
**Pull request by HeavenManySugar**
This pull request adds camera shake functionality to the existing Camera class. It includes the implementation of the Shake method, which allows the camera to shake for a specified duration and strength. The camera shake is triggered when certain events occur in the game, such as the player defeating an enemy or opening a door.

### CODE REVIEW

File: include/Game/Camera.h [hunk 1]
```diff
@@ -52,14 +52,25 @@ class Camera final {
     void MoveByTime(const unsigned long &duringTimeMs,
                     const glm::vec2 &destination);
 
+    void Shake(const unsigned long &duringTimeMs, const float &strength);
```
Comment (onon1101) [inline include/Game/Camera.h:55]: why need to use the references for unsigned long and float?

Comment (HeavenManySugar) [reply include/Game/Camera.h:55]: > why need to use the references for unsigned long and float?

why not?

Comment (HeavenManySugar) [reply include/Game/Camera.h:55]: Depending on the given duration and strength, there will be different shaking effects

Comment (onon1101) [reply include/Game/Camera.h:55]: they are value, not the address. It recommend to use the `pass by value` not `pass by references`

Comment (onon1101) [reply include/Game/Camera.h:55]: if you do something to change the value of `duringTimeMs` in callee, they will also change unwittingly the value of the caller

Comment (onon1101) [reply include/Game/Camera.h:55]: 就是假設你在methods 裡面不小心改到了參數的值, 這會導致在外部呼叫這個方法的變數也一同被改變
這是非常危險的

Comment (onon1101) [reply include/Game/Camera.h:55]: 我不太清楚你這樣做 舉體的意義在哪 如果是為了統一都用references 呼叫的話 不太建議

Comment (HeavenManySugar) [reply include/Game/Camera.h:55]: > 就是假設你在methods 裡面不小心改到了參數的值, 這會導致在外部呼叫這個方法的變數也一同被改變 這是非常危險的

好那我把一些用不到更改的都改成pass by value

Comment (HeavenManySugar) [reply include/Game/Camera.h:55]: 我今天聽到shared_ptr好像一定要用pass by value

Comment (onon1101) [reply include/Game/Camera.h:55]: really?
不過確實 smart ptr 本身就是一種 value , 所以我大部分的 smart ptr 都是用 pass by value

**Review by onon1101 — APPROVED** [review]
OK

**Pull request merged**